src/envoy/core.clj

```diff
@@ -112,23 +109,26 @@
   "Called when a variable is accessed in the environment map with the key and
   original (string) config value. Returns the processed value."
   [k v]
-  ; Update access counter.
+  ;; Update access counter.
   (swap! accesses update k (fnil inc 0))
-  ; Look up variable definition.
+  ;; Look up variable definition.
   (if-let [definition (get known-vars k)]
     (if (some? v)
-      ; Parse the string value with custom parser or for known types.
+      ;; Parse the string value with custom parser or for known types.
       (if-let [parser (:parser definition)]
         (parser v)
         (if-let [type-key (:type definition)]
           (types/parse type-key v)
           v))
       (if-let [default (:default definition)]
-        default
-        ; Check if the var has missing behavior.
+        (do
+          (log/infof "Environment variable %s has no value, using default '%s'"
```

Comment:
Might this be more appropriate to log at the `debug` level?

Reply:
I had it at debug at first. I should only log once per initialization so log volume is not a concern and having it more likely to be present in production environments (with logging level filter higher than debug, presumably) seems worthwhile.